src/i965_drv_video.c

```diff
@@ -2742,6 +2743,15 @@ i965_CreateContext(VADriverContextP ctx,
 
     i965->current_context_id = contextID;
 
+    i965->bufmgr = drm_intel_bufmgr_gem_init(intel->fd, 4096);
+
+    i965->gem_context = drm_intel_gem_context_create(i965->bufmgr);
```

Comment:
drm_intel_gem_context_create might fail too.

Reply:
Ok, will take a closer look.

Reply:
@xhaihao please take a look at updated patch on this branch.